library/spdm_common_lib/libspdm_com_crypto_service.c

```diff
@@ -1047,13 +1047,43 @@ bool libspdm_verify_challenge_auth_signature(libspdm_context_t *spdm_context,
     }
 
     if (is_requester) {
+        /**
+         * No compile time assert in libspdm requires using runtime assert instead.
+         *
+         * To keep the spdm_crypt_lib agnostic of spdm_context types, the "endian"
+         * parameter of spdm_crypt_lib::libspdm_asym_verify(_hash)_ex is type uint32_t.
+         * The type representing the same thing in spdm_context is and enum type of
+         * libspdm_spdm_10_11_verify_signature_endian_t. Generally enum types are the
+         * same size as uint32_t, but there are compile options to shrink enums sizes.
+         *
+         * The first assert guarantees there will be no conflict in the sizes of these
+         * types, as one is used in place of the other. (spdm_context->
+         * spdm_10_11_verify_signature_endian is passed in as endian parameter.)
+         *
+         * The remaining asserts verify the endian values have the same meaning.
+         * Ex. spdm_context::LE == spdm_crypt_lib::LE
+         *     spdm_context::BE == spdm_crypt_lib::BE
+         *     spdm_context::BE or LE == spdm_crypt_lib:: BE or LE
+         *
+         * If any of these asserts pop, be sure to search for other usages of
+         * spdm_context->spdm_10_11_verify_signature_endian and be sure it is correct.
+         */
+        LIBSPDM_ASSERT(sizeof(uint32_t) ==
+                       sizeof(spdm_context->spdm_10_11_verify_signature_endian));
+        LIBSPDM_ASSERT(LIBSPDM_SPDM_10_11_ASYM_VERIFY_FLAG_LITTLE_ENDIAN_ONLY ==
+                       LIBSPDM_SPDM_10_11_VERIFY_SIGNATURE_LITTLE_ENDIAN_ONLY);
+        LIBSPDM_ASSERT(LIBSPDM_SPDM_10_11_ASYM_VERIFY_FLAG_LITTLE_BIG_ONLY ==
+                       LIBSPDM_SPDM_10_11_VERIFY_SIGNATURE_LITTLE_BIG_ONLY);
+        LIBSPDM_ASSERT(LIBSPDM_SPDM_10_11_ASYM_VERIFY_FLAG_BIG_OR_LITTLE_ENDIAN ==
+                       LIBSPDM_SPDM_10_11_VERIFY_SIGNATURE_BIG_OR_LITTLE_ENDIAN);
+
```

Comment:
Note there are two different types here for specifying "endianness". One is in the context and is of type libspdm_spdm_10_11_verify_signature_endian_t. The other is within the spdm_crypt_lib and is a uint32_t with macros for each endian type. The values overlap for each type so you can use one in place of the other. I don't really like this (using 2 types for the same meaning) but I did it this way because the spdm_crypt_lib seems to be agnostic about types defined for the context.  Another option would be to put a single type spdm_10_11_endian types in spdm.h, but I wasn't sure if this ok this feels like a workaround for endianness missing in the spdm 1.0/1.1 spec. If you prefer adding a single type to spdm.h, I can do that.

Reply:
What was wrong with defining the flags in `spdm_crypt_lib.h`?

Reply:
Good point. I thought I would have to pull in another header file into spdm_context, but looks like I don't. I can make defines in spdm_crypt_lib.h and use them in the context files.